src/gtk/textctrl.cpp

```diff
@@ -728,7 +728,21 @@ bool wxTextCtrl::Create( wxWindow *parent,
         gtk_entry_get_text((GtkEntry*)m_text);
 
         if (style & wxNO_BORDER)
-            g_object_set (m_text, "has-frame", FALSE, NULL);
+        {
+#ifdef __WXGTK3__
+            // this is sort of a workaround for when the builtin theme
+            // won't remove the frame by itself -- see
+            // https://bugzilla.gnome.org/show_bug.cgi?id=789732#c1
+            GtkCssProvider* provider = gtk_css_provider_new();
+            std::string rule = !gtk_check_version(3,20,0) ? "entry" : "GtkEntry";
+            rule.append(" { border: none; border-radius: 0; padding: 0 }");
+
+            wxWindowGTK::ApplyCssStyle(provider, rule.c_str());
+            g_object_unref(provider);
```

Comment:
Could we please use `wxGtkObject` from `include/wx/gtk/private/object.h` instead of manual `g_object_unref()`?

This may seem like needless complication for such a simple piece of code, but simple code has tendency to become less simple with time and it could well happen that someone introduces a `return` somewhere above resulting in a memory leak with the code as written.